### PR TITLE
WIP Use mp_trees rather than gb_merkle_trees

### PR DESCRIPTION
**Pull request by gorillainduction**
I label this WIP since I should decide if I should await the database before proceeding.

Please review this as if the WIP is not stated.

Note that this PR is replacing the gb_merkle_trees with the functional version of aeu_mp_trees without using any of the benefits of MPTs. The hashes will be the same when a proper db connection is added, but then we can start rewriting the state handling to make some proper benefit of the MPTs.

### CODE REVIEW

File: deployment/ansible/deploy.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@
           port: 3013
       chain:
         persist: true
-        db_path: "./db15"
+        db_path: "./db16"
```
Comment (lucafavatella) [inline deployment/ansible/deploy.yml]: master has db17, this shall be db18 (just for not forgetting)

File: apps/aeutils/src/aeutils.app.src [hunk 2]
```diff
@@ -9,8 +9,7 @@
     gproc,
     jobs,
     exometer_core,
-    yamerl,
-    gb_merkle_trees
```
Comment (lucafavatella) [inline apps/aeutils/src/aeutils.app.src]: Leftover in rebar.config and rebar.lock

**Comment by gorillainduction** [comment]
Rebased, resolved conflicts, and removed gb_merkle_trees from rebar files.

**Comment by lucafavatella** [comment]
Somehow CI failed https://travis-ci.org/aeternity/epoch/jobs/333260094#L1242

**Comment by gorillainduction** [comment]
I had missed removing a bogus dep stated in aecore.app.src

**Comment by lucafavatella** [comment]
I see both CI builds finished. I see two approvers. Merging soon.

**Pull request merged**